.github/workflows/lint-and-format.yaml

```diff
@@ -0,0 +1,28 @@
+name: "lint-and-format"
+
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the main branch
+  push: 
+    branches: [ develop, develop-0.1, develop-0.2 ]
+  pull_request:
+    branches: [ develop, develop-0.1, develop-0.2 ]
+
+jobs:
+  fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: dtolnay/rust-toolchain@stable
+      
+      - run: cargo fmt --all --check
+
+  clippy:
+    runs-on: ubuntu-latest
+    steps:
+     - uses: actions/checkout@v4
```

Comment:
This checks out the repo and installs rust a second time, should the run step for clippy not be merged into the job above?

Reply:
I separated them for brevity, though merging them wouldn't also be a bad idea, although logs would need to be checked to know whether it was clippy or fmt that failed

Reply:
Fixed, and merged into the `test.yaml` workflow